llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -8308,6 +8367,115 @@ llvm::Function *SPIRVToLLVM::createLibraryEntryFunc() {
   return func;
 }
 
+const Vkgc::PipelineOptions *SPIRVToLLVM::getPipelineOptions() const {
+  assert(m_context && "Invalid context!");
+
+  return static_cast<Llpc::Context *>(m_context)->getPipelineContext()->getPipelineOptions();
+}
+
+template <> SPIRVValue *SPIRVToLLVM::getValueForScratchBoundaryCheckWrapper<SPIRVStore>(SPIRVStore *toBeWrapped) const {
+  return toBeWrapped->getDst();
+}
+
+template <> SPIRVValue *SPIRVToLLVM::getValueForScratchBoundaryCheckWrapper<SPIRVLoad>(SPIRVLoad *toBeWrapped) const {
+  return toBeWrapped->getSrc();
+}
+
+template <class T>
+std::pair<bool, SPIRVToLLVM::ScratchBoundaryCheckInfo> SPIRVToLLVM::prepareScratchBoundaryCheck(T *origin) {
```

Comment:
This function is very long but has no documentation and examples. Could you explain what it does and show an example?